### PR TITLE
Update One Box Code Snippets UI

### PR DESCRIPTION
**Pull request by thenamankumar**
closes: https://linear.app/sourcegraph/issue/SRCH-1100/one-box-search-improvements-correct-issues-from-dog-fooding
closes: https://linear.app/sourcegraph/issue/SRCH-1115/make-individual-search-results-in-one-box-collapsible

- Bring back the result hide functionality
- Include the ask Cody button at bottom of file
- Limit displayed results to 5, with show more button
- File path background should be opaque.

Designs: https://www.figma.com/design/eR495BJclVpdGdXf4lhSsH/One-box?node-id=632-7402&node-type=canvas&t=gs8ttscPyqpA2YTj-0

Loom: https://www.loom.com/share/1c1d01bc6f734eee8ff0f54cae8adf96

## Test plan

- Enable Onebox and execute code search intent
- Make sure only first 5 results are shown and "show more" and "ask the llm" button are shown at the bottom.
- Make sure the code snippets now have the dark bg same as the code snippets returned in the Cody response. 
- The file link does not have any border now. 
- Test the hide code snippet function by clicking on the caret left to the file link.

## Changelog

### CODE REVIEW

**Comment by rrhyne** [comment]
Hi Naman, we get a loom and or some screenshots on this change?

File: vscode/webviews/chat/cells/contextCell/ContextCell.tsx [hunk 1]
```diff
@@ -171,29 +178,55 @@ export const ContextCell: FunctionComponent<{
...
+                                                contextItemsToDisplay.length > 5 ? (
+                                                    <div className="tw-flex tw-justify-between">
+                                                        <Button
+                                                            variant="link"
+                                                            onClick={() => setShowAllResults(true)}
+                                                        >
+                                                            Show {contextItemsToDisplay.length - 5} more
+                                                            results
+                                                        </Button>
+                                                        <Button
```
Comment (vovakulikov) [inline vscode/webviews/chat/cells/contextCell/ContextCell.tsx:219]: Should we render this resubmit button always, or only when we have more than 5 results?

File: vscode/webviews/components/codeSnippet/CodeSnippet.module.css [hunk 2]
```diff
@@ -55,7 +49,13 @@
 }
 
 .result {
-    background-color: var(--cody-chat-code-background);
+    border-radius: 4px;
+    border: solid 1px var(--cody-chat-code-border-color);
+    background-color: var(--coyd-chat-code-background) !important;
```
Comment (vovakulikov) [inline vscode/webviews/components/codeSnippet/CodeSnippet.module.css:54]: Have you found some problems with CSS cascade that you need `!important ` statement here?

File: vscode/webviews/components/codeSnippet/components/FileMatchChildren.tsx [hunk 3]
```diff
@@ -38,23 +39,23 @@ export const FileMatchChildren: FC<PropsWithChildren<FileMatchProps>> = props =>
     }
 
     const navigateToFile = useCallback(
-        (event: KeyboardEvent<HTMLElement> | MouseEvent<HTMLElement>): void => {
-            // TODO: implement navigation by blob
-            // navigateToCodeExcerpt(event, props.openInNewTab ?? false, navigate)
+        (line: number) => {
+            getVSCodeAPI().postMessage({
```
Comment (vovakulikov) [inline vscode/webviews/components/codeSnippet/components/FileMatchChildren.tsx:43]: This wouldn't work for Cody Web case

File: vscode/webviews/components/codeSnippet/CodeSnippet.module.css (hunk 2 above)
Comment (vovakulikov) [reply vscode/webviews/components/codeSnippet/CodeSnippet.module.css:54]: Also, there is a typo `--coyd-chat-code-background`

**Review by vovakulikov — COMMENTED** [review]
Left a few comments. 

Also noticed that the collapsable block header looks a bit off at the moment (border radius is missing and I think it should be within code snippet border)
 
Screenshot 2024-09-30 at 11 58 31

File: vscode/webviews/components/codeSnippet/components/FileMatchChildren.tsx (hunk 3 above)
Comment (thenamankumar) [reply vscode/webviews/components/codeSnippet/components/FileMatchChildren.tsx:43]: Yes, this won't, I tested on the web, it doesn't break stuff but obviously it doesn't work. For now we can keep it as a IDE only feature until, we figure out how to solve it without rendering a ahref. 

I'll add a todo comment as before in the code for now.

File: vscode/webviews/chat/cells/contextCell/ContextCell.tsx (hunk 1 above)
Comment (thenamankumar) [reply vscode/webviews/chat/cells/contextCell/ContextCell.tsx:219]: CleanShot 2024-09-30 at 21 15 48@2x

Changed the BG color. It needs to be outside the border based on the designs.

File: vscode/webviews/components/codeSnippet/CodeSnippet.module.css (hunk 2 above)
Comment (thenamankumar) [reply vscode/webviews/components/codeSnippet/CodeSnippet.module.css:54]: Thanks for the catch.

File: vscode/webviews/components/codeSnippet/components/FileMatchChildren.tsx (hunk 3 above)
Comment (thenamankumar) [reply vscode/webviews/components/codeSnippet/components/FileMatchChildren.tsx:44]: added comment

**Comment by thenamankumar** [comment]
@rrhyne here you go: https://www.loom.com/share/1c1d01bc6f734eee8ff0f54cae8adf96

**Review by vovakulikov — APPROVED** [review]

**Pull request merged**